corehq/apps/accounting/static/accounting/js/accounting.software_plan_version_handler.js

```diff
@@ -1,437 +1,359 @@
-            if (self.new.hasMatchingRole() && self.isRoleTypeNew()) {
-                self.existing.roleSlug(self.new.matchingRole().slug());
-                $('#id_role_slug').select2('val', self.new.matchingRole().slug());
-            }
-        })
-    };
-};
-
-var NewRoleManager = function (existingRoles, newPrivileges) {
-    'use strict';
-    var self = this;
-
-    self.existingRoles = existingRoles;
-
-    self.privileges = ko.observableArray(newPrivileges);
-    self.matchingRole = ko.computed(function () {
-        // If the set of current privileges match the privileges of an existing role, return that existing role.
-        var existingRole = null;
-        _.each(self.existingRoles(), function (role) {
-            var isEquivalent = _.isEmpty(role.privilegeSlugs()) && _.isEmpty(self.privileges());
-            if (isEquivalent || (!_.isEmpty(role.privilegeSlugs())
-                    && _.isEmpty(_(role.privilegeSlugs()).difference(self.privileges()))
-                    && role.privilegeSlugs().length === self.privileges().length)) {
-                existingRole = role;
+        self.matchingPrivileges = ko.computed(function () {
+            if (self.matchingRole()) {
+                return self.matchingRole().privileges();
             }
+            return [];
         });
-        return existingRole;
-    });
-    self.matchingPrivileges = ko.computed(function () {
-        if (self.matchingRole()) {
-            return self.matchingRole().privileges();
-        }
-        return [];
-    });
-    self.allowCreate = ko.computed(function () {
-        return !_.isEmpty(self.privileges()) && _.isEmpty(self.matchingPrivileges());
-    });
-    self.hasMatchingRole = ko.computed(function () {
-        return !_.isNull(self.matchingRole());
-    });
-};
-
-
-var ExistingRoleManager = function (existingRoles, currentRoleSlug) {
-    'use strict';
-    var self = this;
-
-    self.existingRoles = existingRoles;
-
-    self.roleSlug = ko.observable(currentRoleSlug);
-    self.selectedRole = ko.computed(function () {
-        var selectedRole = null;
-        _.each(self.existingRoles(), function (role) {
-            if (role.slug() === self.roleSlug()) {
-                selectedRole = role;
-            }
+        self.allowCreate = ko.computed(function () {
+            return !_.isEmpty(self.privileges()) && _.isEmpty(self.matchingPrivileges());
+        });
+        self.hasMatchingRole = ko.computed(function () {
+            return !_.isNull(self.matchingRole());
         });
-        return selectedRole;
-    });
-    self.selectedPrivileges = ko.computed(function () {
-        if (self.selectedRole()) {
-            return self.selectedRole().privileges();
-        }
-        return [];
-    });
-    self.hasNoPrivileges = ko.computed(function () {
-        return _.isEmpty(self.selectedPrivileges())
-    });
-};
-
-
-var BaseSelect2Handler = function (options, currentSelectionObj) {
```

Comment:
Yeah, that was my reading - that it was copy-pasted with some tweaks at some point, but the code eventually changed so those tweaks aren't necessary any more.